index.js

```diff
@@ -0,0 +1,348 @@
+'use strict';
+
+const Scm = require('screwdriver-scm-base');
+const async = require('async');
+const hoek = require('hoek');
+
+class ScmRouter extends Scm {
+    /**
+     * Constructs a router for different scm strategies
+     * @method constructor
+     * @param  {Object}         config                             Object with scms and ecosystem
+     * @param  {Object}         [config.ecosystem]                 Optional object with ecosystem values
+     * @param  {Object}         config.scms                        Array of scms to load or a single scm object
+     * @param  {String}         config.scms[x].plugin              Name of the scm NPM module to load
+     * @param  {String}         config.scms[x].config              Configuration to construct the module with
+     * @param  {String}         config.scms[x].config.displayName  Nickname to displaoy of the scm
+     * @return {ScmRouter}
+     */
+    constructor(config = {}) {
+        const ecosystem = config.ecosystem;
+        const scmsConfig = config.scms;
+
+        super();
+
+        this.scms = {};
+
+        if (typeof scmsConfig === 'object') {
+            if (!(Array.isArray(scmsConfig))) {
+                throw new Error('No scm config passed in.');
```

Comment:
I'm not sure about this line because from the description for the param `config.scms` it sounds like it can either be an array of scms to load or a single scm object. So if a single scm object is passed wouldn't this throw an error? Is this intended behavior?